src/ui.form-radio/URadio.vue

```diff
@@ -1,167 +1,34 @@
 
 import ULabel from "../ui.form-label/ULabel.vue";
 import { getDefault } from "../utils/ui.ts";
 
-import defaultConfig from "./config.js";
-import useAttrs from "./useAttrs.js";
-import { URadio } from "./constants.js";
+import defaultConfig from "./config.ts";
+import useAttrs from "./useAttrs.ts";
+import { URadio } from "./constants.ts";
+
+import type { URadioProps } from "./types.ts";
+import type { SetRadioGroupSelectedItem } from "../ui.form-radio-group/types.ts";
 
 defineOptions({ inheritAttrs: false });
 
-const setRadioGroupSelectedItem = inject("setRadioGroupSelectedItem", null);
+const setRadioGroupSelectedItem = inject<SetRadioGroupSelectedItem | null>(
+  "setRadioGroupSelectedItem",
+  null,
+);
 const getRadioGroupName = inject("getRadioGroupName", null);
 const getRadioGroupColor = inject("getRadioGroupColor", null);
 const getRadioGroupSize = inject("getRadioGroupSize", null);
 const getRadioGroupSelectedItem = inject("getRadioGroupSelectedItem", null);
 
-const props = defineProps({
-  /**
-   * Radio value.
-   */
-  modelValue: {
-    type: [Boolean, String, Number, Array, Object],
-    default: null,
-  },
-
-  /**
-   * Native value attribute.
-   */
-  value: {
-    type: [Boolean, String, Number, Array, Object],
-    default: "",
-  },
-
-  /**
-   * Radio label.
-   */
-  label: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Label placement.
-   * @values left, right
-   */
-  labelAlign: {
-    type: String,
-    default: getDefault(defaultConfig, URadio).labelAlign,
-  },
-
-  /**
-   * Radio description.
-   */
-  description: {
-    type: [String, Object],
-    default: "",
-  },
-
-  /**
-   * Error message.
-   */
-  error: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Radio name.
-   */
-  name: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Radio size.
-   * @values sm, md, lg
-   */
-  size: {
-    type: String,
-    default: getDefault(defaultConfig, URadio).size,
-  },
-
-  /**
-   * Radio color.
-   * @values brand, grayscale, gray, red, orange, amber, yellow, lime, green, emerald, teal, cyan, sky, blue, indigo, violet, purple, fuchsia, pink, rose
-   */
-  color: {
-    type: String,
-    default: getDefault(defaultConfig, URadio).color,
-  },
-
-  /**
-   * Set radio disabled.
-   */
-  disabled: {
-    type: Boolean,
-    default: getDefault(defaultConfig, URadio).disabled,
-  },
-
-  /**
-   * Set radio checked.
-   */
-  checked: {
-    type: Boolean,
-    default: getDefault(defaultConfig, URadio).checked,
-  },
-
-  /**
-   * Unique element id.
-   */
-  id: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Component config object.
-   */
-  config: {
-    type: Object,
-    default: () => ({}),
-  },
-
-  /**
-   * Data-test attribute for automated testing.
-   */
-  dataTest: {
-    type: String,
-    default: "",
-  },
+const props = withDefaults(defineProps<URadioProps>(), {
```

Comment:
Declare default modelValue as it was before.

Reply:
Default value was "null". Do we really need to declare default modelValue now?

Reply:
Otherwise it will be undefined when v-model is not provided. It's fine in this component, but i think it may break some other components where undefined is not expected, so it's better to leave default value.